cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/server.go

```diff
@@ -194,11 +197,14 @@ func (s *Server) GetAvailablePackageSummaries(ctx context.Context, request *core
 		repo := v1alpha1.PackageRepository{
 			Name: name,
 		}
-		// namespace is optional according to https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/
 		namespace, found, err := unstructured.NestedString(obj, "metadata", "namespace")
-		if err == nil && found {
-			repo.Namespace = namespace
+		if err != nil || !found {
+			// should not happen in reality
+			log.Errorf("field metadata.namespace not found on HelmRepository: %w:\n%v", err, obj)
+			// just skip over to the next one
+			continue
 		}
+		repo.Namespace = namespace
 
 		repoPackages, err := readPackagesFromRepoIndex(&repo, url)
```

Comment:
Probably worth a TODO at the start of the surrounding iteration that each repo should be read in a separate go routine, if this code is used in the future.

Reply:
will do